eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -199,10 +201,43 @@ class NodeRelay private(nodeParams: NodeParams,
             context.log.warn(s"rejecting trampoline payment reason=$failure")
             rejectPayment(upstream, Some(failure))
             stopping()
+          case None =>
+            if (nextPayload.isAsyncPayment && nodeParams.features.hasFeature(Features.AsyncPaymentPrototype)) {
+              waitForTrigger(upstream, nextPayload, nextPacket)
+            } else {
+              doSend(upstream, nextPayload, nextPacket)
+            }
+        }
+      case RelayAsyncPayment => context.log.debug("received async payment trigger while waiting to receive all incoming payment packets")
+        receiving(htlcs, nextPayload, nextPacket, handler)
+    }
+
+  private def waitForTrigger(upstream: Upstream.Trampoline, nextPayload: IntermediatePayload.NodeRelay.Standard, nextPacket: OnionRoutingPacket): Behavior[Command] = {
+    context.log.info(s"waiting for async payment to trigger before relaying trampoline payment (amountIn=${upstream.amountIn} expiryIn=${upstream.expiryIn} amountOut=${nextPayload.amountToForward} expiryOut=${nextPayload.outgoingCltv}, timeout=${nodeParams.relayParams.timeout})")
+    Behaviors.withTimers { timers =>
+      timers.startSingleTimer(AsyncPaymentTimeout, nodeParams.relayParams.timeout)
+      Behaviors.receiveMessagePartial {
+        case AsyncPaymentTimeout =>
+          context.log.warn(s"rejecting async payment that was not triggered before relay timeout: ${nodeParams.relayParams.timeout}")
+          rejectPayment(upstream, Some(PaymentTimeout))
```

Comment:
We shouldn't use `PaymentTimeout`, this tell the sender that we didn't receive their incoming HTLCs, whereas the issue here is that the recipient is unavailable. Right now for Phoenix when this happens, we end up using `TemporaryNodeFailure`. This isn't ideal, but it's the best existing failure message that matches this. We should add a dedicated new failure code for this scenario, but I believe we should only do this for the final specification, for now `TemporaryNodeFailure` is a good enough choice.

Reply:
fixed in [fcdc394](https://github.com/ACINQ/eclair/pull/2435/commits/fcdc394cb2751d7408318afb437cf49d18389d08).